aviary/examples/external_subsystems/engine_NPSS/table_engine_builder.py

```diff
@@ -0,0 +1,186 @@
+                          engine_data[:, 3],
+                          units='lbf',
+                          desc='Current net thrust produced')
+        engine.add_output(Dynamic.Mission.THRUST_MAX,
+                          thrust_max,
+                          units='lbf',
+                          desc='Max net thrust produced')
+        engine.add_output(Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE,
+                          -engine_data[:, 4],
+                          units='lbm/s',
+                          desc='Current fuel flow rate ')
+        engine.add_output(Dynamic.Mission.ELECTRIC_POWER,
+                          zeros_array,
+                          units='kW',
+                          desc='Current electric energy rate')
+        engine.add_output(Dynamic.Mission.NOX_RATE,
+                          zeros_array,
+                          units='lb/h',
+                          desc='Current NOx emission rate')
+        engine.add_output(Dynamic.Mission.TEMPERATURE_ENGINE_T4,
+                          zeros_array,
+                          units='degR',
+                          desc='Current turbine exit temperature')
+        return engine
+
+    def get_bus_variables(self):
+      # Transfer training data from pre-mission to mission
+        return vars_to_connect
+
+    def get_controls(self, phase_name):
+        '''
+        Builds dictionary of controls for engine off-design.
+
+        Returns
+        -------
+        controls : dict
+            Dictionary with keys that are names of variables to be controlled and the
+            values are dictionaries with the keys `units`, `upper`, and `lower` which states the units of the
+            variable to be controlled.
+        '''
+
+        # TODO: we will need to handle this throttle_dict differently if the
+        # injected electric input magnitude is greater than 400 hp
+        # throttle_dict = {'units': None, 'upper': 1, 'lower': .625, 'ref': 1}
+
+        # HPX_dict = {'units': 'hp', 'upper': 0., 'lower': -400, 'ref': 400., 'opt': True}  # bounds are notional for now
+
+        # if phase_name == 'climb':
+        #    controls_dict = {
+        #        Dynamic.Engine.ELECTRIC_SHAFT_POWER : HPX_dict,
+        #        }
+        # else:
+        #    controls_dict = {}
+
+        # return controls_dict
+        return {}
+
+    def get_design_vars(self):
+        '''
+        Builds dictionary of design variables for Engine off-design.
+
+        Returns
+        -------
+        design_vars : dict
+            Dictionary with keys that are names of variables to be made design variables
+            and the values are dictionaries with the keys `units`, `upper`, `lower`, and `ref`.
+        '''
+        mass_flow_dict = {'units': 'lbm/s', 'upper': 450, 'lower': 100,
+                          'ref': 450}  # upper and lower are just notional for now
+        # GTOW_dict = {'units': 'kg', 'upper': 70_000, 'lower': 60_000, 'ref': 70_000}
```

Comment:
Is GTOW from an example problem, or does NPSS sometimes need it? I think you can remove it, since it is commented out.

Reply:
removed and uploaded